### PR TITLE
CA Calworks childcare and CAPP

### LINKED ISSUE #7196: CA Calworks childcare and CAPP

### PR DESCRIPTION
**Pull request by hua7450**
Fixes #7196

## Summary

Implements California CalWORKs Stage 2, Stage 3, and California Alternative Payment Program (CAPP) child care subsidies.

## Regulatory Authority

| Source | Description |
|--------|-------------|
| [WIC § 10370-10376.5](https://leginfo.legislature.ca.gov/faces/codes_displayText.xhtml?lawCode=WIC&division=9.&title=&part=1.8.&chapter=2.&article=) | CalWORKs child care program (Stages 1, 2, 3) |
| [WIC § 10271](https://leginfo.legislature.ca.gov/faces/codes_displaySection.xhtml?sectionNum=10271.&lawCode=WIC) | General child care eligibility, CAPP |
| [WIC § 10271.5](https://leginfo.legislature.ca.gov/faces/codes_displaySection.xhtml?sectionNum=10271.5.&lawCode=WIC) | Income eligibility (85% SMI) |
| [WIC § 10290](https://leginfo.legislature.ca.gov/faces/codes_displaySection.xhtml?sectionNum=10290.&lawCode=WIC) | Family fee schedule |
| [WIC § 10372](https://leginfo.legislature.ca.gov/faces/codes_displaySection.xhtml?sectionNum=10372.&nodeTreePath=16.4.19&lawCode=WIC) | Stage 2 eligibility |
| [WIC § 10372.5](https://leginfo.legislature.ca.gov/faces/codes_displaySection.xhtml?sectionNum=10372.5.&nodeTreePath=16.4.19&lawCode=WIC) | Stage 3 eligibility |
| [5 CCR § 18078](https://www.law.cornell.edu/regulations/california/5-CCR-18078) | Countable income definition |
| [AB 116 (2023)](https://leginfo.legislature.ca.gov/faces/billTextClient.xhtml?bill_id=202320240AB116) | Family fee reform (75% SMI exemption, 1% cap) |

## Program Structure

California subsidized child care operates in stages based on CalWORKs history:

| Stage | Eligibility | Legal Citation |
|-------|-------------|----------------|
| **Stage 1** | Currently receiving CalWORKs cash aid | WIC § 10370 |
| **Stage 2** | Former CalWORKs within 24 months of leaving cash aid | WIC § 10372 |
| **Stage 3** | Former CalWORKs who exhausted Stage 2 eligibility (>24 months) | WIC § 10372.5 |
| **CAPP** | Income-eligible families who **never** received CalWORKs AND not currently receiving TANF | WIC § 10271 |

### Program Mutual Exclusivity

Programs are mutually exclusive - a household can only qualify for ONE program at a time:

| Scenario | ca_tanf > 0 | was_calworks | months_since | Eligible Program |
|----------|:-----------:|:------------:|:------------:|:----------------:|
| Currently on TANF | ✓ | any | N/A | Stage 1 |
| Former, ≤24 months | ✗ | ✓ | ≤24 | Stage 2 |
| Former, >24 months | ✗ | ✓ | >24 | Stage 3 |
| Never CalWORKs | ✗ | ✗ | N/A | CAPP |

## Income Eligibility

| Test | Threshold | Legal Citation |
|------|-----------|----------------|
| **Stage 2/3/CAPP eligibility** | ≤ 85% SMI | WIC § 10271.5, AB 99 (2017) |

### Countable Income (5 CCR § 18078)
Counted: employment income, self-employment income, unemployment compensation, workers' compensation, state disability insurance, social security (excluding SSI), child support, alimony, pension income, veterans' benefits, interest, dividends, capital gains, rental income.

**Not counted:** SSI (federal program, explicitly excluded)

## Family Fees

| Income Level | Fee | Legal Citation |
|--------------|-----|----------------|
| Below 75% SMI | **Exempt** | WIC § 10290, AB 116 (2023) |
| At/above 75% SMI | **1% of monthly income** | WIC § 10290, AB 116 (2023) |
| Stage 1 (on CalWORKs) | **Exempt** | Categorical exemption |

## Benefit Calculation

```
Payment = min(Regional Market Rate Ceiling, Actual Child Care Cost) - Family Fee
```

RMR ceilings vary by child age, provider type, and time category.

### New Variables
- `ca_child_care_smi` - State Median Income calculation
- `ca_child_care_countable_income` - Per 5 CCR § 18078 (excludes SSI)
- `ca_child_care_income_eligible` - 85% SMI income test
- `ca_child_care_family_fee` - Family fee calculation (75% exemption, 1% cap)
- `ca_calworks_stage_2_eligible`, `ca_calworks_stage_2_payment`, `ca_calworks_stage_2`
- `ca_calworks_stage_3_eligible`, `ca_calworks_stage_3_payment`, `ca_calworks_stage_3`
- `ca_capp_eligible`, `ca_capp_payment`, `ca_capp`
- `was_calworks_recipient` - Input: ever received CalWORKs cash aid
- `months_since_calworks_exit` - Input: months since leaving cash aid

### New Parameters
- `gov.states.ca.cdss.child_care.income.income_limit` (0.85)
- `gov.states.ca.cdss.child_care.income.sources` (list of countable income types)
- `gov.states.ca.cdss.child_care.family_fees.exemption_threshold` (0.75)
- `gov.states.ca.cdss.child_care.family_fees.fee_cap` (0.01)
- `gov.states.ca.cdss.child_care.stage_2.time_limit_months` (24)
- `gov.hhs.smi.additional_person_threshold` (6) - Parameterized SMI household size adjustment threshold

### Modified Variables
- `ca_child_care_subsidies` - Added Stage 2, Stage 3, and CAPP (changed to MONTH period)
- `ca_capp_eligible` - Added check for not currently receiving TANF (prevents double counting)

### Integration Tests (9 scenarios)
1. Stage 1 family currently on CalWORKs
2. Stage 2 former CalWORKs within 24 months
3. Stage 3 former CalWORKs beyond 24 months
4. CAPP for families who never received CalWORKs
5. Stage 2 with family fee applied (income above 75% SMI)
6. Income too high for any program (above 85% SMI)
7. Two children with different age-based RMR ceilings
8. YEAR period test for Stage 2
9. **Minimal inputs test** - verifies basic inputs work with sensible defaults (CAPP path)

### Edge Case Tests (16 cases)
- **Stage 2**: 24-month boundary (eligible), zero months since exit, minimal TANF ($0.01 still disqualifies)
- **Stage 3**: 25-month boundary (eligible), 24-month boundary (not eligible - still Stage 2)
- **Income**: zero income eligible, exactly at 85% SMI, above 85% SMI
- **Family fee**: zero income (exempt), exactly at 75% SMI, below 75% SMI (exempt)
- **SMI**: 2-person, 6-person, 7-person (tests household size adjustments)
- **CAPP**: adult-only household (no children), zero income (fails work requirement)

### Mutual Exclusivity Verified
All scenarios result in exactly ONE eligible program with no double counting:
- Currently on TANF → Stage 1 only
- Former CalWORKs ≤24 months → Stage 2 only
- Former CalWORKs >24 months → Stage 3 only
- Never CalWORKs, not on TANF → CAPP only
- First-time TANF enrollee → Stage 1 only (not CAPP)

## Minimum Required Inputs

For a basic simulation with non-zero subsidy:
```yaml
people:
  adult:
    age: 30
    employment_income: 36_000  # Work requirement
  child:
    age: 3
    pre_subsidy_childcare_expenses: 24_000
    ca_calworks_child_care_time_category: MONTHLY  # Required for non-zero coefficient
households:
  household:
    state_code: CA
```

With these inputs and defaults (`was_calworks_recipient: false`, `ca_tanf: 0`), the household qualifies for CAPP.

### CODE REVIEW

File: policyengine_us/variables/gov/states/ca/cdss/child_care/capp/ca_capp_eligible.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from policyengine_us.model_api import *
+
+
+class ca_capp_eligible(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ca/cdss/child_care/capp/ca_capp_eligible.py:4]: Can we check in an integration test with basic inputs income, and avoid double counting capp and calworks

File: policyengine_us/variables/gov/states/ca/cdss/child_care/common/ca_child_care_smi.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        four_person_smi = parameters(instant_str).gov.hhs.smi.amount[
+            state_code
+        ]
+        adjustment = parameters(
+            instant_str
+        ).gov.hhs.smi.household_size_adjustment
+        size_adjustment = (
+            adjustment.first_person
+            + adjustment.second_to_sixth_person * (min_(size, 6) - 1)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ca/cdss/child_care/common/ca_child_care_smi.py]: make 6 a parameter as max_unit_size or something a long those lines

File: policyengine_us/variables/gov/states/ca/cdss/child_care/common/ca_child_care_smi.py [hunk 3]
```diff
@@ -22,15 +22,13 @@ def formula(spm_unit, period, parameters):
...
 
-        four_person_smi = parameters(instant_str).gov.hhs.smi.amount[
-            state_code
-        ]
-        adjustment = parameters(
-            instant_str
-        ).gov.hhs.smi.household_size_adjustment
+        p = parameters(instant_str).gov.hhs.smi
+        four_person_smi = p.amount[state_code]
+        adjustment = p.household_size_adjustment
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ca/cdss/child_care/common/ca_child_care_smi.py]: use these directly in the formula

File: policyengine_us/variables/gov/states/ca/cdss/child_care/common/ca_child_care_smi.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        state_code = spm_unit.household("state_code_str", period.this_year)
+
+        p = parameters(instant_str).gov.hhs.smi
+        threshold = p.additional_person_threshold
+        size_adjustment = (
+            p.household_size_adjustment.first_person
+            + p.household_size_adjustment.second_to_sixth_person
+            * (min_(size, threshold) - 1)
+            + p.household_size_adjustment.additional_person
+            * max_(size - threshold, 0)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ca/cdss/child_care/common/ca_child_care_smi.py]: break out formula

File: policyengine_us/parameters/gov/states/ca/cdss/child_care/income/sources.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    - child_support_received
+    - alimony_income
+    # Retirement and veterans
+    - pension_income
+    - veterans_benefits
+    # Investment income
+    - interest_income
+    - dividend_income
+    - capital_gains
+    - rental_income
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ca/cdss/child_care/income/sources.yaml:26]: Check if we can dd ca_tanf - if not leave reference for circular reference

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/states/ca/cdss/child_care/common/ca_child_care_smi.py (hunk 3 above)
Comment (hua7450) [reply policyengine_us/variables/gov/states/ca/cdss/child_care/common/ca_child_care_smi.py]: Have a better approach in https://github.com/PolicyEngine/policyengine-us/pull/7172, will apply in this PR as well.

File: policyengine_us/parameters/gov/states/ca/cdss/child_care/income/sources.yaml (hunk 5 above)
Comment (hua7450) [reply policyengine_us/parameters/gov/states/ca/cdss/child_care/income/sources.yaml:26]: Yes we could, does not make any difference though, because if ca_tanf > 0, it must be stage 1.
